Tests/Unit/MSFT_xDnsServerForwarder.Tests.ps1

```diff
@@ -0,0 +1,112 @@
+    InModuleScope $Global:DSCResourceName {
+
+        #region Pester Test Initialization
+        $forwarders = '192.168.0.1','192.168.0.2'
+        $testParams = @{
+            IsSingleInstance = 'Yes'
+            IPAddresses = $forwarders
+        }
+        $fakeCimInstance = @{
+            Forwarders = $forwarders
+        }
+        #endregion
+
+
+        #region Function Get-TargetResource
+        Describe "$($Global:DSCResourceName)\Get-TargetResource" {
+            It 'Returns a "System.Collections.Hashtable" object type' {
+                Mock -CommandName Get-CimInstance -MockWith {return $fakeCimInstance}
+                $targetResource = Get-TargetResource @testParams
+                $targetResource -is [System.Collections.Hashtable] | Should Be $true
+            }
+
+            It "Returns IPAddresses = $($testParams.IPAddresses) when forwarders exist" {
+                Mock -CommandName Get-CimInstance -MockWith {return $fakeCimInstance}
+                $targetResource = Get-TargetResource @testParams
+                $targetResource.IPAddresses | Should Be $testParams.IPAddresses
+            }
+
+            It "Returns an empty IPAddresses when forwarders don't exist" {
+                Mock -CommandName Get-CimInstance -MockWith {}
+                $targetResource = Get-TargetResource @testParams
+                $targetResource.IPAddresses | Should Be $null
+            }
+        }
+        #endregion
+
+
+        #region Function Test-TargetResource
+        Describe "$($Global:DSCResourceName)\Test-TargetResource" {
+            It 'Returns a "System.Boolean" object type' {
+                Mock -CommandName Get-CimInstance -MockWith {return $fakeCimInstance}
+                $targetResource =  Test-TargetResource @testParams
+                $targetResource -is [System.Boolean] | Should Be $true
+            }
+
+            It 'Passes when forwarders match' {
+                Mock -CommandName Get-CimInstance -MockWith {return $fakeCimInstance}
+                Test-TargetResource @testParams | Should Be $true
+            }
+
+            It "Fails when forwarders don't match" {
+                Mock -CommandName Get-CimInstance -MockWith {}
+                Test-TargetResource @testParams | Should Be $false
+            }
+        }
+        #endregion
+
+
+        #region Function Set-TargetResource
+        Describe "$($Global:DSCResourceName)\Set-TargetResource" {
+            It "Calls Set-CimInstance once" {
+                Mock -CommandName Set-CimInstance -MockWith {}
+                Set-TargetResource @testParams
+                Assert-MockCalled -CommandName Set-CimInstance -Times 1 -Exactly -Scope It
+            }
+        }
+        #endregion
+
+        # TODO: Pester Tests for any Helper Cmdlets
+
+    }
+    #endregion
+}
+finally
+{
+    #region FOOTER
+    Restore-TestEnvironment -TestEnvironment $TestEnvironment
+    #endregion
+
+    # TODO: Other Optional Cleanup Code Goes Here...
+}
```

Comment:
I won't block the merge for this issue alone, but please be aware of this in the future (look at the PR for the symbol you should see here at the end of the diff.)
Files must end with a new line.  See [Style Guidelines, General Rule, #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)